src/ILCompiler.CppCodeGen/src/CppCodeGen/ILToCppImporter.cs

```diff
@@ -279,7 +279,8 @@ private string GetStackValueKindCPPTypeName(StackValueKind kind, TypeDesc type =
                 case StackValueKind.Int64: return "int64_t";
                 case StackValueKind.NativeInt: return "intptr_t";
                 case StackValueKind.ObjRef: return "void*";
-                case StackValueKind.Float: return "double";
+                case StackValueKind.Float:
```

Comment:
The problem occurs when we use reflection to get float fields. Compiler generates following code to unbox float value:
```cpp
double unboxed = *(double*)(float*)((void**)boxed + 1)
```
that produce incorrect result.

Reply:
This should be rather fixed in the implementation of unbox instead of here. 

I think that the same problem will happen for other types too. For example, if the type is `sbyte` and the boxed value is `-1`, we will end up with `0xFF` pushed on the IL stack today. Instead, we should end up with `-1` on the IL stack.

Reply:
Thank you for suggestion!